### ISSUE TITLE
yamllint doesn't ignore shebang lines with whitespace after the shebang marker

### ISSUE THREAD

**Issue by tamere-allo-peter** [adrienverge/yamllint#428]
Whitespace after the _#!_ marker on shebang lines is authorized and optional, as explained on [Wikipedia's entry for shebang line](https://en.wikipedia.org/wiki/Shebang_(Unix)) as can be seen from the extracts below : 

**_White space after #! is optional_**

and 

**_It has been claimed[20] that some old versions of Unix expect the normal shebang to be followed by a space and a slash (#! /), but this appears to be untrue;[21] rather, blanks after the shebang have traditionally been allowed, and sometimes documented with a space_**

However yamllint fails to ignore such shebang lines as can be seen with the following YAML file

```
#! /usr/bin/env blah
---
name: blah
```

which gives this result with _yamllint_'s default configuration 

```
stdin:1:2: [warning] missing starting space in comment (comments)
```

This issue seems to be related to the presence of '\S' in the regex used to match the shebang lines in _comments.py_

A  simpler condition only veryfying the presence of the _!_ character at the start of the buffer would probably fix this.

I'll try to send a PR.

**Comment by adrienverge** [comment]
Thanks, it's interesting and I wasn't aware about this specificity.

Instead of modifying yamllint's behavior for `ignore-shebangs` option (which could lead to unhappy users), I'd rather accept a PR that allows more flexibility. Could this be solved by adding an `ignore-regex` option, as suggested in https://github.com/adrienverge/yamllint/issues/361 and https://github.com/adrienverge/yamllint/issues/416?

```yaml
rules:
  comments:
    ignore-regex:
      - ^#!\s*
      - ^#cloud-config
```

**Comment by tamere-allo-peter** [comment]
Not sure I'll be able to do this soon. I would have to do a more in-depth study of yamllint I suppose, but this will have to wait until next week at least. This was more a quick week-end hack BTW ;-)
